conformance/features/httpgateway/httpgateway.go

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package httpgateway
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/cucumber/godog"
+	"github.com/cucumber/messages-go/v10"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	tstate "sigs.k8s.io/gateway-api/conformance/state"
+	"sigs.k8s.io/gateway-api/conformance/utils"
+)
+
+var (
+	state *tstate.Scenario
+)
+
+// InitializeScenario configures the Feature to test
+func InitializeScenario(ctx *godog.ScenarioContext) {
+	ctx.Step(`^a new random namespace$`, aNewRandomNamespace)
+	ctx.Step(`^the "([^"]*)" scenario$`, theScenario)
+	ctx.Step(`^Gateway.Status.Conditions "([^"]*)" condition should be set to "([^"]*)" within (\d+) minutes$`, conditionMustBeTrueWithin)
+
+	ctx.BeforeScenario(func(*godog.Scenario) {
+		state = tstate.New()
+	})
+
+	ctx.AfterScenario(func(*messages.Pickle, error) {
+		// delete namespace an all the content
+		_ = utils.DeleteNamespace(state.Namespace)
+	})
+}
+
+func aNewRandomNamespace() error {
+	ns, err := utils.NewNamespace()
+	if err != nil {
+		return err
+	}
+
+	state.Namespace = ns
+	return nil
+}
+
+func theScenario(name string) error {
+	dp := utils.DynamicParams{
+		Path:      name + ".yaml",
+		Namespace: state.Namespace,
+	}
+	return utils.DynamicApply(dp)
+}
+
+func conditionMustBeTrueWithin(condName string, condValue string, minutes int) error {
```

Comment:
Can we/do we want to have this expose more info? Like condition not found vs False, etc?

This exact case its probably not that important, just wondering in general

Reply:
Yeah I experimented with this, it's not quite as straightforward as `t.Logf`, but using `klog` seems acceptable here.